src/components/options/SpcOptionEditor.tsx

```diff
@@ -28,24 +28,32 @@ export function SpcOptionEditor(props: Props) {
   const styles = useStyles2(getStyles);
 
   const { value: options, isVar } = useParseSpcOptions(props.value);
+  const [customOptions, setCustomOptions] = React.useState<Array<SelectableValue<number>>>([]);
   const onChange = props.onChange;
 
   return (
     <>
       <div className={styles.row}>
         <InlineField label="Sample size" disabled={isVar}>
           <Select
-            options={sampleSizeOptions}
+            options={[...sampleSizeOptions, ...customOptions]}
+            placeholder={options.sampleSize.toString()}
             value={options.sampleSize}
-            onChange={(e) => {
-              const newSampleSize = e.value ?? 1;
+            onChange={(v) => {
+              const newSampleSize = v.value ?? 1;
               const newSpc: SpcOptions = { ...options, sampleSize: newSampleSize };
               if (newSampleSize === 1) {
                 newSpc.aggregation = 'mean';
               }
               onChange({ ...options, sampleSize: newSampleSize });
             }}
             width={'auto'}
+            allowCustomValue={true}
+            onCreateOption={(v) => {
+              const customValue: SelectableValue<number> = { value: parseInt(v, 10), label: v.toString() };
```

Comment:
What will happen when the user enters a string that is not a number?
What if parseInt returns NaN?